src/sentry/search/snuba/executors.py

```diff
@@ -280,32 +283,42 @@ def snuba_search(
             turbo=get_sample,  # Turn off FINAL when in sampling mode
             sample=1,  # Don't use clickhouse sampling, even when in turbo mode.
         )
-
-        error_conditions = self.update_conditions(
-            "event.type",
-            "!=",
-            "transaction",
-            organization_id,
-            project_ids,
-            environments,
-            environment_ids,
-            conditions,
-        )
-        snuba_error_results = query_partial(
-            conditions=error_conditions,
-            aggregations=aggregations,
-            condition_resolver=snuba.get_snuba_column_name,
-        )
-
-        mod_agg = aggregations.copy() if aggregations else []
-        mod_agg.insert(0, ["arrayJoin", ["group_ids"], "group_id"])
-
-        rows = snuba_error_results["data"]
-        total = snuba_error_results["totals"]["total"]
-        row_length = len(rows)
+        group_categories = set()
+
+        for group_type in group_types:
+            for category, types in get_category_type_mapping().items():
+                if GroupType(group_type) in types:
+                    group_categories.add(category)
+
+        rows = []
+        total = 0
+        row_length = 0
+        if not group_categories or GroupCategory.ERROR in group_categories:
+            error_conditions = self.update_conditions(
+                "event.type",
+                "!=",
+                "transaction",
+                organization_id,
+                project_ids,
+                environments,
+                environment_ids,
+                conditions,
+            )
+            snuba_error_results = query_partial(
+                conditions=error_conditions,
+                aggregations=aggregations,
+                condition_resolver=snuba.get_snuba_column_name,
+            )
+            rows = snuba_error_results["data"]
+            total = snuba_error_results["totals"]["total"]
+            row_length = len(rows)
 
         organization = Organization.objects.get(id=organization_id)
-        if features.has("organizations:performance-issues", organization):
+        if (
+            features.has("organizations:performance-issues", organization)
+            and not group_categories
+            or GroupCategory.PERFORMANCE in group_categories
```

Comment:
We need parens here `(not group_categories or GroupCategory.PERFORMANCE in group_categories)`